src/VisualStudio/IntegrationTest/IntegrationTests/VisualBasic/BasicLineCommit.cs

```diff
@@ -0,0 +1,137 @@
+
+namespace Roslyn.VisualStudio.IntegrationTests.VisualBasic
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class BasicLineCommit : AbstractEditorTest
+    {
+        protected override string LanguageName => LanguageNames.VisualBasic;
+
+        public BasicLineCommit(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory, nameof(BasicLineCommit))
+        {
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.LineCommit)]
+        void CaseCorrection()
+        {
+            VisualStudio.Editor.SetText(@"Module Foo
+    Sub M()
+Dim x = Sub()
+    End Sub
+End Module");
+
+            VisualStudio.Editor.PlaceCaret("Sub()", charsOffset: 1);
+            VisualStudio.Editor.SendKeys(VirtualKey.Enter);
+            VisualStudio.Editor.Verify.CaretPosition(48);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.LineCommit)]
+        void UndoWithEndConstruct()
+        {
+            VisualStudio.Editor.SetText(@"Module Module1
+    Sub Main()
+    End Sub
+    REM
+End Module");
+
+            VisualStudio.Editor.PlaceCaret("    REM");
+            VisualStudio.Editor.SendKeys("sub", VirtualKey.Escape, " foo()", VirtualKey.Enter);
+            VisualStudio.Editor.Verify.TextContains(@"Sub foo()
+
+    End Sub");
+            VisualStudio.ExecuteCommand(WellKnownCommandNames.Edit_Undo);
+            VisualStudio.Editor.Verify.CaretPosition(54);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.LineCommit)]
+        void UndoWithoutEndConstruct()
+        {
+            VisualStudio.Editor.SetText(@"Module Module1
+
+    ''' <summary></summary>
+    Sub Main()
+    End Sub
+End Module");
+
+            VisualStudio.Editor.PlaceCaret("Module1");
+            VisualStudio.Editor.SendKeys(VirtualKey.Down, VirtualKey.Enter);
+            VisualStudio.Editor.Verify.TextContains(@"Module Module1
+
+
+    ''' <summary></summary>
+    Sub Main()
+    End Sub
+End Module");
+            VisualStudio.Editor.Verify.CaretPosition(18);
+            VisualStudio.ExecuteCommand(WellKnownCommandNames.Edit_Undo);
+            VisualStudio.Editor.Verify.CaretPosition(16);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.LineCommit)]
+        void CommitOnSave()
+        {
+            VisualStudio.Editor.SetText(@"Module Module1
+    Sub Main()
+    End Sub
+End Module");
+
+            VisualStudio.Editor.PlaceCaret("(", charsOffset: 1);
+            VisualStudio.Editor.SendKeys("x   as   integer", VirtualKey.Tab);
+            VisualStudio.ExecuteCommand("File.SaveSelectedItems");
+            VisualStudio.Editor.Verify.TextContains(@"Sub Main(x As Integer)");
+            VisualStudio.ExecuteCommand(WellKnownCommandNames.Edit_Undo);
+            VisualStudio.Editor.Verify.TextContains(@"Sub Main(x   As   Integer)");
+            VisualStudio.Editor.Verify.CaretPosition(18);
+            VisualStudio.Editor.Verify.CaretPosition(16);
```

Comment:
What does Verify.CaretPosition doing that you can do it twice like this?? Can you add a comment to explain?